### PR TITLE
feat: publish worker sdk starter and docs

### PR DESCRIPTION
**Pull request by sepa79**
## Summary
- add a worker-sdk module that packages control-plane auto-configuration and test fixtures
- document how to bootstrap workers/managers with the new descriptors, emitters, and starter
- update the refactor plan with API availability notes and migration checklist

## Testing
- mvn -pl common/worker-sdk test

------
https://chatgpt.com/codex/tasks/task_e_68dd3d201ca48328bc554045c2a06deb

### CODE REVIEW

File: docs/control-plane-refactor-plan.md [hunk 1]
```diff
@@ -47,9 +47,33 @@ The refactor is technically feasible but high-effort:
 - Keep the new abstractions modular so teams can opt into DSL factories without losing the ability to customise edge-case workflows.
 
 ## Next Steps Checklist
-- [ ] Design topology descriptor interfaces and enumerate required routes/bindings per role.
-- [ ] Implement payload builders and publishers with contract tests.
-- [ ] Provide Spring-friendly factories/starter modules.
+- [x] Design topology descriptor interfaces and enumerate required routes/bindings per role.
+- [x] Implement payload builders and publishers with contract tests.
+- [x] Provide Spring-friendly factories/starter modules.
 - [ ] Migrate one worker service as a pilot and document the process.
```
Comment (sepa79) [inline docs/control-plane-refactor-plan.md:53]: check code, it should be done

File: docs/control-plane-refactor-plan.md [hunk 2]
```diff
@@ -47,9 +47,33 @@ The refactor is technically feasible but high-effort:
...
 - [ ] Roll out to remaining workers, then managers, retiring bespoke helpers.
```
Comment (sepa79) [inline docs/control-plane-refactor-plan.md:54]: check code, it should be done
